tests/reader/reader_test.go

```diff
@@ -1,79 +1,165 @@
-		FilePaths: []string{file1.Name(), file2.Name()},
-	}
-	contentReader := reader.ContentReader{Config: config}
+const (
+	ErrFileNotFound       = "file does not exist or can't be accessed"
+	ErrReadingDirectories = "reading from directories is not currently supported"
+	ErrPermissionDenied   = "permission denied"
+)
 
-	result, err := contentReader.ReadAll()
-	assert.NoError(t, err)
-	expectedResult := content1 + "\n" + content2 + "\n"
-	assert.Equal(t, expectedResult, result)
-}
+func TestReadAll(t *testing.T) {
+	t.Run("reads single file", func(t *testing.T) {
+		file := tests.CreateTempFile(t, tests.SampleText32)
 
-func TestReadFile(t *testing.T) {
-	content := "file content"
-	file, err := tests.CreateTempFile(t, content)
-	if err != nil {
-		t.Fatalf("Failed to create temporary file: %s", err)
-	}
+		config := &options.Config{
+			FilePaths: []string{file.Name()},
+		}
 
-	config := &options.Config{}
-	contentReader := reader.ContentReader{Config: config}
+		contentReader := reader.ContentReader{Config: config}
+
+		result, err := contentReader.ReadAll()
+		require.NoError(t, err)
+		expectedResult := tests.SampleText32 + "\n"
+		require.Equal(t, expectedResult, result)
+	})
+
+	t.Run("reads from argument", func(t *testing.T) {
+		config := &options.Config{
+			Text: tests.SampleText32,
+		}
+
+		contentReader := reader.ContentReader{Config: config}
 
-	result, err := contentReader.ReadFile(file.Name())
-	assert.NoError(t, err)
-	assert.Equal(t, content, result)
+		result, err := contentReader.ReadAll()
+		require.NoError(t, err)
+		expectedResult := tests.SampleText32 + "\n"
+		require.Equal(t, expectedResult, result)
+	})
+
+	t.Run("reads multiple files", func(t *testing.T) {
+		fileContents := []string{"Japan", "Australia", "Germany"}
+		expectedResults := strings.Join(fileContents, "\n") + "\n"
+		var filePaths []string
+
+		for index := 0; index < 3; index++ {
+			file := tests.CreateTempFile(t, fileContents[index])
+
+			filePaths = append(filePaths, file.Name())
+		}
+
+		config := &options.Config{FilePaths: filePaths}
+		contentReader := reader.ContentReader{Config: config}
+
+		result, err := contentReader.ReadAll()
+
+		require.NoError(t, err)
+		require.Equal(t, expectedResults, result)
+	})
 }
 
+//nolint:funlen // It cannot be shorter
 func TestReadable(t *testing.T) {
-	file, err := tests.CreateTempFile(t, "content")
-	if err != nil {
-		t.Fatalf("Failed to create temporary file: %s", err)
-	}
-
 	config := &options.Config{}
 	contentReader := reader.ContentReader{Config: config}
 
-	err = contentReader.Readable(file.Name())
-	assert.NoError(t, err)
+	t.Run("file exists and is readable", func(t *testing.T) {
+		file := tests.CreateTempFile(t, "content")
+		err := contentReader.Readable(file.Name())
+		require.NoError(t, err)
+	})
+
+	t.Run("file does not exist", func(t *testing.T) {
+		err := contentReader.Readable("nonexistentfile")
+		require.Error(t, err)
+		assert.EqualError(t, err, ErrFileNotFound)
```

Comment:
Yes, I noticed it, not sure why I kept the `require.Error()` plus the `assert.EqualError()` together, maybe because I thought it was safer, but now I see it isn't since I'm able to check for the error once with `require.ErrorIs()`, simpler and safer